pkg/io/applicator.go

```diff
@@ -120,8 +122,37 @@ func (a *APIApplicator) Apply(ctx context.Context, current client.Object, opts .
 	if err != nil {
 		return fmt.Errorf("converting desired obj to unstructured: %w", err)
 	}
-	unstructured.RemoveNestedField(before, "status")
-	unstructured.RemoveNestedField(after, "status")
+
+	// remove the status if it's a subresource
+	statusFieldPath, err := fieldpath.MakePath("status")
+	if err != nil {
+		return fmt.Errorf("error creating field path: %w", err)
+	}
+	hasStatusSubresource := false
+	for _, managedFields := range current.GetManagedFields() {
+		// we're doing a client-side apply, so we assume we own all fields even if the manager is not our own.
+		// in other words, no need to ensure that managedFields.Manager == a.managerName
+		// TODO: we should explore using server-side apply
+		if managedFields.Subresource == "" {
+			// if the field is managed by root, then it's not a status subresource
+			continue
+		}
+		var fieldPaths fieldpath.Set
+		if managedFields.FieldsType != "FieldsV1" {
+			return fmt.Errorf("unsupported fields type %v", managedFields.FieldsType)
+		}
+		err := fieldPaths.FromJSON(bytes.NewReader(managedFields.FieldsV1.Raw))
+		if err != nil {
+			return fmt.Errorf("error parsing managedFields for %v: %w", m, err)
+		}
+		if fieldPaths.Has(statusFieldPath) {
+			hasStatusSubresource = true
+		}
+	}
```

Comment:
```suggestion
	hasStatusSubresource := false
	for _, managedFields := range current.GetManagedFields() {
		if managedFields.Subresource == "status" {
			hasStatusSubresource = true
			break
		}
	}
	// remove the status if it's a subresource
```

Reply:
I think we can just check, for each entry in `metadata.managedFields`, whether `.subresource: status` exists

Reply:
🔕 addressed in https://github.com/reddit/achilles-sdk/pull/17/commits/dc5aa4e0ed7179d23344f2b1bf2cc786b9db2d09